src/components/side-panel/side-panel-header.tsx

```diff
@@ -18,21 +22,41 @@ export const SidePanelHeader = () => {
         togglePanel,
         removeFocusedElement,
         focusedElement,
+        updateElement,
+        updateFocusedElement,
     } = useBuilder();
 
+    const { reset, handleSubmit } = useFormContext<IBaseTextBasedFieldValues>();
+
     // =========================================================================
     // EVENT HANDLERS
     // =========================================================================
 
-    // TODO: When react hook form is being set up, run this function only when there are changes in the form values
-    // const handleSaveButtonClick  = () => {
-    //     if (onSaveChanges) onSaveChanges();
-    //     return "Save button clicked";
-    // };
     const handleCrossButtonClick = () => {
         removeFocusedElement();
     };
 
+    const onSubmit = useCallback(
+        (values) => {
+            console.log("values:", values);
+            updateElement(values);
+            updateFocusedElement(false, values);
+            reset(
+                {
+                    ...values,
+                },
+                {
+                    keepDirty: false,
+                }
+            );
```

Comment:
can follow the implementation here: https://stackoverflow.com/a/76741818
but implement this in `SidePanel` instead
so the submit handler here should be preventing the native form submission via `event.preventDefault()`

```
const {
    formState: { isSubmitSuccessful },
    reset,
    handleSubmit,
} = useFormContext<IBaseTextBasedFieldValues>();

useEffect(() => {
    reset(undefined, {
        keepValues: true,
        keepDirty: false,
    });
}, [isSubmitSuccessful]);
```